airflow/config_templates/config.yml

```
@@ -2591,3 +2591,25 @@ sensors:
       type: float
       example: ~
       default: "604800"
+telemetry_collection:
+  description: |
+    Airflow integrates `Scarf <https://about.scarf.sh/>`__ to collect basic telemetry data during operation.
+    This data assists Airflow maintainers in better understanding how Airflow is used.
+    Insights gained from this telemetry are critical for prioritizing patches, minor releases, and
+    security fixes. Additionally, this information supports key decisions related to the development road map.
+    Check the FAQ doc for more information on what data is collected.
+
+    Deployments can opt-out of analytics by setting the ``enabled`` option
+    to ``False``, or the ``SCARF_ANALYTICS=false`` environment variable.
```

I don't see that you modify anything in the config loading logic, following the other options the correct ENV must be named...
```suggestion
    to ``False``, or the ``AIRFLOW__TELEMETRY_COLLECTION__ENABLED=False`` environment variable.
```

---

This is the alternate option. I want to respect what Scarf has in their docs to disable too

https://github.com/apache/airflow/blob/ab1855a8d3a367f5ce36bf25a72ffebfe31448f7/airflow/settings.py#L579-L583